lib/solidus_kustomer/client.rb

```diff
@@ -34,14 +34,70 @@ def create(kobject_klass, attributes:, customer:)
       response.success? || raise(SolidusKustomer::CreateError)
     end
 
-    # Finds the customer via the Solidus database id
+    # Finds a customer via their Spree::User record. If the user does not have a kustomer_id
+    # associated, it also updates the original record to cache the UUID.
     #
-    # @param user [Spree::User] the user whose data we're looking for
+    # @param user [Spree::User] the customer we're looking for
     #
-    # @return [Hash] the customer data
-    def find_customer_by_external_id(user)
+    # @return [Hash] the customer data or nil if not present
+    def find_customer(user)
+      return find_customer_by_uuid(user.kustomer_id) if user.kustomer_id
+
+      customer = find_customer_by_email(user.email) || find_customer_by_external_id(user.id)
+      user.update!(kustomer_id: customer['id']) if customer
+      customer
+    end
+
+    # Finds a customer UUID via their Spree::User record. If the user does not have a kustomer_id
+    # associated, it also updates the original record to cache the UUID.
+    #
+    # @param user [Spree::User] the customer's UUID we're looking for
+    #
+    # @return [String] the customer UUID or nil if not present
+    def find_customer_id(user)
```

Comment:
You should rename this method to make it clear that it doesn't just find a customer, it also sets the Kustomer ID on the respective `Spree::User` record.

Reply:
I've gone with a descriptive yet longish name, such as `find_customer_id_and_update_user_if_necessary`. I don't like it very much but it's explicit.